mobile/test/common/integration/xds_test_server_interface.cc

```diff
@@ -0,0 +1,54 @@
+#include "test/common/integration/xds_test_server_interface.h"
+
+#include "test/common/integration/xds_test_server.h"
+
+#include "extension_registry.h"
+
+// NOLINT(namespace-envoy)
+
+static std::shared_ptr<Envoy::XdsTestServer> strong_test_server_;
+static std::weak_ptr<Envoy::XdsTestServer> weak_test_server_;
```

Comment:
curious, why do we need the weak_ptr? and why are both these variables static?

Reply:
It's `static` because it's going to be used by JNI with C linkage (`extern "C"`) and C has no namespace. The use of `weak_ptr` is so that we can pass the non-owning reference to other functions without increasing the reference counter. The other reason is mainly because I just followed the existing style 😄 
  https://github.com/envoyproxy/envoy/blob/67e4108c3a78c75cfc2237f876a316115429e1de/mobile/test/common/integration/test_server_interface.cc#L15